src/printer.js

```diff
@@ -1666,6 +1666,9 @@ function genericPrintNoParens(path, options, print, args) {
       if (n.type === "TSAbstractClassProperty") {
         parts.push("abstract ");
       }
+      if (n.readonly || hasReadonly(n, options)) {
```

Comment:
I think that it would be safer to check from `start(n)` to `start(n.key)` instead of the entire node, which could be big.